src/Npgsql/TypeHandlers/TextHandler.cs

```diff
@@ -231,7 +231,7 @@ public virtual int ValidateAndGetLength(ArraySegment<char> value, ref NpgsqlLeng
             if (parameter?.Size > 0)
                 throw new ArgumentException($"Parameter {parameter.ParameterName} is of type ArraySegment<char> and should not have its Size set", parameter.ParameterName);
 
-            return lengthCache.Set(_encoding.GetByteCount(value.Array, value.Offset, value.Count));
+            return lengthCache.Set(_encoding.GetByteCount(value.Array!, value.Offset, value.Count));
```

Comment:
Can ArraySegment.Array really be null, or is it just badly annotated in corefx? If it's the former we should probably do a check...

Reply:
It's a bug. Will fix it in a separate PR to backport it.